rpc/providers/proxy.ts

```diff
@@ -4,11 +4,22 @@ import * as B from "../Base.ts";
 import { FailedToSendMessageError, ParseRawIngressMessageError } from "../common.ts";
 import { WsContainer } from "./ws_container.ts";
 
-export async function proxyClient<Config_ extends Config<string>>(
+export async function proxyClient<Config_ extends Config<string | string[]>>(
   config: Config_,
 ): Promise<ProxyClient<Config_> | FailedToOpenConnectionError> {
+  const createWebSocketFactory = () => {
+    let index = 0;
+    return (discoveryValue: Config_["discoveryValue"]) => {
+      if (typeof discoveryValue === "string") {
+        return new WebSocket(discoveryValue);
+      }
+      index %= discoveryValue.length;
+      return new WebSocket(discoveryValue[index++]!);
+    };
+  };
   const ws = new WsContainer({
-    webSocketFactory: () => new WebSocket(config.discoveryValue),
+    discoveryValue: config.discoveryValue,
+    webSocketFactory: createWebSocketFactory(),
```

Comment:
This will work for reconnects but not for initial connections.

This makes me think that, it would be easier to model the reconnect as a stream of attempts to reconnect.
Once the attempts are exhausted, the following should happen
- if it's creating the `WsContainer`, it should fail with `FailedToOpenConnectionError`
- if there are messages waiting for a connection, it should fail with `FailedToSendMessageError`

The above could be simplified if a connection attempt is only triggered when there is a message to be sent.

Reply:
cc @harrysolovay